Libraries/Azure.psm1

```diff
@@ -1821,9 +1821,15 @@ Function Delete-ResourceGroup([string]$RGName, [bool]$UseExistingRG, [string]$Pa
     Function RemoveAzResourcesByResourceType([string] $ResourceType4iMatch) {
         $unlockedResources | Where-Object {$_.ResourceType -imatch $ResourceType4iMatch} | ForEach-Object {
             try {
-                $null = Remove-AzResource -ResourceGroupName $RGName -ResourceName $_.Name -ResourceType $_.ResourceType -Force -Verbose
+                $removeResult = Remove-AzResource -ResourceGroupName $RGName -ResourceName $_.Name -ResourceType $_.ResourceType -Force
+                if ($removeResult -eq $true) {
+                    Write-LogInfo "Deleted resource '$($_.Name)' successfully."
+                }
+                else {
+                    Write-LogWarn "Failed to delete resource '$($_.Name)', will try to remove it in next attempt $($attempts + 1)"
```

Comment:
If this has the same error string, can use 'throw'?

Reply:
the try catch is making sure function robustness, usually, at least current version of Remove-AzResource for powershell 5.1 will return false when failed, but not sure the future behavior, and this is the end of each run-end of test case, no outer logic will and can help on such of removing, also it can recover by itself if complex dependencies are not removed (that's why we use attempts). 
Changed a little bit about the messages used.

Reply:
Looks ok.